src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -359,22 +372,23 @@ chip::ChipError::StorageType pychip_ReadClient_ReadAttributes(void * appContext,
         ReadPrepareParams params(session.Value());
         params.mpAttributePathParamsList    = readPaths.get();
         params.mAttributePathParamsListSize = n;
-
+        params.mIsFabricFiltered            = pyParams.isFabricFiltered;
         if (pyParams.isSubscription)
         {
             params.mMinIntervalFloorSeconds   = pyParams.minInterval;
             params.mMaxIntervalCeilingSeconds = pyParams.maxInterval;
+            err                               = readClient->SendAutoResubscribeRequest(std::move(params));
+        }
+        else
+        {
+            err = readClient->SendRequest(params);
         }
-
-        params.mIsFabricFiltered = pyParams.isFabricFiltered;
-
-        err = readClient->SendRequest(params);
-        SuccessOrExit(err);
     }
 
     *pReadClient = readClient.get();
     *pCallback   = callback.get();
 
+    readPaths.release();
```

Comment:
Isn't this a leak in the non-subscription case?

Reply:
Fixed, thanks